src/openMVG/matching_image_collection/gpu/CudaBruteForceMatcher.cu

```diff
@@ -0,0 +1,66 @@
+/*******************************************************************
```

Comment:
As we have discussed with Kareem we can license the matcher under the MPL2 License:

Let's convert the top header to
```
// This file is part of OpenMVG, an Open Multiple View Geometry C++ library.

// Copyright (c) 2016 Kareem Omar (kareem.omar@uah.edu - https://github.com/komrad36)

// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.

// Fastest GPU implementation of a brute-force
// Hamming-weight matrix for 512-bit binary descriptors.
//
/ Yes, that means the DIFFERENCE in popcounts is used
// for thresholding, NOT the ratio. This is the CORRECT
// approach for binary descriptors.
//
// This laboriously crafted kernel is EXTREMELY fast.
// 43 BILLION comparisons per second on a stock GTX1080,
// enough to match nearly 38,000 descriptors per frame at 30 fps (!)
//
// A key insight responsible for much of the performance of
// this insanely fast CUDA kernel is due to
// Christopher Parker (https://github.com/csp256), to whom
// I am extremely grateful.
//
// CUDA CC 3.0 or higher is required.
```